### PR TITLE
Add local checkpoints and pre-commit copy check

### PR DESCRIPTION
**Pull request by pstjohn**
Adds a local checkpoint to the recipe directories that's used in unit tests to let folks more easily debug issues in the model's forward pass, and adds a pre-commit check to ensure that these files don't drift out-of-date with the model definition in models/esm2

## Summary by CodeRabbit

- New Features
  - Added TransformerEngine-optimized ESM-2 models and Masked LM variants with local 8M example checkpoints, configs, tokenizer, vocab, and special tokens.
  - Training configs updated to reference local example checkpoints for easier local testing.

- Documentation
  - Added README files for example checkpoints and Apache 2.0 LICENSE files.

- Chores
  - Added a pre-commit hook and a utility to verify/synchronize copied files.
  - Updated lint configuration and per-file ignores.

### CODE REVIEW

File: bionemo-recipes/recipes/esm2_native_te/example_8m_checkpoint/tokenizer_config.json [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+      "single_word": false,
+      "special": true
+    }
+  },
+  "clean_up_tokenization_spaces": false,
+  "cls_token": "<cls>",
+  "eos_token": "<eos>",
+  "extra_special_tokens": {},
+  "mask_token": "<mask>",
+  "model_max_length": 1000000000000000019884624838656,
```
Comment (trvachov) [inline bionemo-recipes/recipes/esm2_native_te/example_8m_checkpoint/tokenizer_config.json:49]: what is this number?

File: bionemo-recipes/recipes/esm2_native_te/example_8m_checkpoint/esm_nv.py [hunk 2]
```diff
@@ -0,0 +1,601 @@
...
+        self.padded_vocab_size = padded_vocab_size if padded_vocab_size is not None else self.vocab_size
+
+        # Ensure padded_vocab_size is at least as large as vocab_size
+        if self.padded_vocab_size is not None and self.vocab_size is not None:
+            assert self.padded_vocab_size >= self.vocab_size, (
+                f"padded_vocab_size ({self.padded_vocab_size}) must be greater than or equal to vocab_size ({self.vocab_size})"
+            )
+
+
+class NVEsmEncoder(nn.Module):
```
Comment (trvachov) [inline bionemo-recipes/recipes/esm2_native_te/example_8m_checkpoint/esm_nv.py:116]: This is duplicate code to the model, is that right?

File: bionemo-recipes/recipes/esm2_accelerate_te/example_8m_checkpoint/README.md [hunk 3]
```diff
@@ -0,0 +1,8 @@
+# Example 8M ESM-2 Checkpoint
+
+This directory contains the model and tokenizer configuration and forward pass code for an ESM-2 8M model, identical to
+huggingface.co/nvidia/esm2_t6_8m_ur50d. This directory can be useful for testing modifications to the model's forward
+pass, and allows the unit tests to run without requiring external HuggingFace API calls.
+
```
Comment (trvachov) [inline bionemo-recipes/recipes/esm2_accelerate_te/example_8m_checkpoint/README.md:8]: In both READMEs, could you add a small 1-2 sentences about how this looks like when you pull from HuggingFace? I know this is essentially documenting HF cache, but if you briefly write it here it would help justify why these checkpoints exist.

File: bionemo-recipes/recipes/esm2_native_te/hydra_config/defaults.yaml [hunk 4]
```diff
@@ -1,5 +1,5 @@
 # Training config
-model_tag: ???
+model_tag: ??? # E.g., nvidia/esm2_t6_8M_UR50D or facebook/esm2_t6_8M_UR50D, or a local path.
```
Comment (trvachov) [inline bionemo-recipes/recipes/esm2_native_te/hydra_config/defaults.yaml]: could you put `./example_8m_checkpoint` after `local path`? e.g. `or a local path such as ./example_8m_checkpint` ?

File: bionemo-recipes/recipes/esm2_native_te/example_8m_checkpoint/tokenizer_config.json (hunk 1 above)
Comment (pstjohn) [reply bionemo-recipes/recipes/esm2_native_te/example_8m_checkpoint/tokenizer_config.json:49]: No idea, never actually looked at this file before -- but it comes from upstream https://huggingface.co/facebook/esm2_t6_8M_UR50D/blob/main/tokenizer_config.json

I'm guessing it means we don't want to be limited by the tokenizer in generating proteins

File: bionemo-recipes/recipes/esm2_native_te/example_8m_checkpoint/esm_nv.py (hunk 2 above)
Comment (pstjohn) [reply bionemo-recipes/recipes/esm2_native_te/example_8m_checkpoint/esm_nv.py:116]: Yeah that's right, this is the model's forward pass. It would actually get run during unit tests though, so `breakpoint()` etc. will work here.
